controller/voucher.js

```diff
@@ -0,0 +1,144 @@
+            new_voucher.discount = req.body.discount;
+            code = code;
+
+            new_voucher.save((err) => {
+                if(!err)
+                res.json('Saved Successfully');
+                else {
+                        res.json('Error occur when saving : ' + err);   
+                    
+                }
+            });
+        }
+
+        router.get('/unique-voucher/:id',  (req, res) => {
+            insertUserVoucher
+        });
+
+        function insertUserVoucher(req, res){
+            if(!req.body.recepient || !req.body.email ){
+                res.json({
+                    success: false,
+                    message: 'recepient and offer required'
+                })
+            }else{
+            Recepient.findOne({email : req.body.email}).then(recepient =>{
+            var id = recepient.id
+        new_voucher.findOne({code: req.body.code, recepient: id}, (err, voucher)=>{
+            var expire = new Date(voucher.expiryDate)
+            var currentDate = new Date()
+            if(!voucher.used && currentDate > expire ){
+                voucher.comparecode(req.body.code, function(err, match) {
+                    if (match && !err) {
+                      var offer = voucher.offer
+                      offer.findById(req.params.id , (err, res) =>{
+                          if(err){
+                            res.json({
+                                success: false,
+                                message: 'offer not found'
+                            })
+                        }
+                        else{
+                            Voucher.findByIdAndUpdate({_id: voucher._id}, {$set: {track: new Date(), used: true}}, (err, track)=>{
+                                if(err){
+                                  res.json({
+                                      success: true,
+                                      message: 'unable to update voucher track',
+                                      
+                                  })
+                                }else{
+                                  res.json({
+                                      success: true,
+                                      message: 'found discount',
+                                      discount: res.discount
+                                  })
+                                }
+                            })
+                          
+                        }
+                    })
+                } else {
+                    res
+                      .status(401)
+                      .send({ success: false, message: "Incorrect voucher code!!" });
+                  }
+                });
+          }
+      })
+      
+      }).catch(err =>{
+          res.json({
+              success: false,
+              message: 'recepient not found'
+              
+          })
+      })
+      
+  }
+
+        })
+
+
+module.exports = router;
```

Comment:
the code in the file is hard to read, you need to move out your logic into separate file. There is I chance that a bug is in here, I'm even sure there is. please fix.